pandas/tests/indexes/test_numeric.py

```diff
@@ -259,6 +259,12 @@ def test_astype(self):
         for dtype in ['M8[ns]', 'm8[ns]']:
             self.assertRaises(TypeError, lambda: i.astype(dtype))
 
+        # GH 13149
+        # a NaN astype int
```

Comment:
Done.

(BTW, DatetimeIndex.astype(int), TimedeltaIndex.astype(int) return a numpy.array. Isn't it a bug?  PeriodIndex and Float64Index return Int64Index.)